src/main/java/uk/gov/hmcts/probate/schedule/HmrcExtractTask.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.probate.schedule;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.probate.service.dataextract.DataExtractDateValidator;
+import uk.gov.hmcts.probate.service.dataextract.HmrcDataExtractService;
+import uk.gov.hmcts.reform.probate.model.client.ApiClientException;
+
+import java.time.LocalDate;
+
+import static uk.gov.hmcts.probate.model.Constants.DATE_FORMAT;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class HmrcExtractTask implements Runnable {
+
+    private final DataExtractDateValidator dataExtractDateValidator;
+    private final HmrcDataExtractService hmrcDataExtractService;
+
+    @Value("${adhocSchedulerJobDate}")
+    public String adHocJobFromDate;
+    @Value("${adhocSchedulerJobToDate}")
+    public String adHocJobToDate;
+    private String toDate;
```

Comment:
not sure you need this (toDate) to be a class variable

Reply:
Changed